src/application/service_tester.go

```diff
@@ -141,11 +144,38 @@ func (s *ServiceTester) setup() error {
 
 // Run runs the tests for the service and return true if the tests passed
 // and an error if any
-func (s *ServiceTester) Run() (int, error) {
+func (s *ServiceTester) Run() (types.TestResult, error) {
 	if err := s.setup(); err != nil {
-		return 1, err
+		return types.TestResult{}, err
+	}
+
+	var isInterrupted bool
+	var err error
+	wg := new(sync.WaitGroup)
+	wg.Add(1)
+	go func() {
+		c := make(chan os.Signal, 1)
+		signal.Notify(c, os.Interrupt)
+		<-c
+		signal.Stop(c)
+		err = s.Shutdown()
+		isInterrupted = true
+		wg.Done()
+	}()
```

Comment:
I don't understand how this works `wait group` works as we never wait on it. Can we add a feature test for this where the test prints "started", sleeps for a minute, and prints "finished". We hit ctrl-c after "started" is seen, and check that the containers are shut down, we don't see finished, and no tests run, and it exits wth status 1?

Reply:
I'm not sure exactly where `wg.Wait()` should go, or even `wg.Add(1)`. In `exo run` we would wait indefinitely until the user hit ctrl-c but in exo-clean that's not the case, so I'm not sure where to increment the counter of the wait group

Reply:
Please write a test first. I'm uncertain what is necessary here. I don't think we need the wait group either way. One thing though is that the goroutine needs to communicate with the main thread via a channel. We may need to move the `runTests` into a goroutine as well and if either complete, they pass something into the channel which the main thread reads from:

```go
done := make(chan TestResult)
// create a goroutine listening for sigInt, if caught push a result into channel
// create a goroutine running the tests, if finished push result into channel
return <- done // wait for one to finish
```
To get the error in case runTests hits an error, we may need a custom struct that contains a testResult and an error.